cftime/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+//! Time handling according to CF conventions
+#![allow(missing_docs)]
+
+use nom::{
+    branch::alt,
+    bytes::complete::tag,
+    character::complete::{i32, i8, one_of, space1, u32, u8},
+    combinator::{all_consuming, map, opt, peek, value},
+    number::complete::double,
+    sequence::{preceded, separated_pair, tuple},
+    IResult,
+};
+
+/// Parsing error
+#[derive(Debug)]
+pub struct ParseError(String);
+
+impl std::fmt::Display for ParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl std::error::Error for ParseError {}
+
+/// Base duration between time points
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub enum Duration {
+    Years,
+    Months,
+    Days,
+    Hours,
+    Minutes,
+    Seconds,
+    Milliseconds,
+    Microseconds,
+}
```

Comment:
This implementation is not trivial since Month depends on Years which depends on the calendar. Maybe need something like this :
```rust
pub enum Calendars {}

impl Duration {
    fn get_seconds(calendar:Calendars){
        // Some implementations
    }
}
```

Reply:
Yeah, this will be quite the tricky per-calendar stuff